src/app/base/components/MachineLink/MachineLink.tsx

```diff
@@ -17,13 +14,9 @@ export enum Labels {
 }
 
 const MachineLink = ({ systemId }: Props): JSX.Element | null => {
-  const machine = useSelector((state: RootState) =>
-    machineSelectors.getById(state, systemId)
-  );
-  const machinesLoading = useSelector(machineSelectors.loading);
-  useFetchMachines();
+  const { machine, loading } = useGetMachine(systemId);
 
-  if (machinesLoading) {
+  if (loading) {
```

Comment:
It looks like you might need to update the test to add a details factory, like this:

https://github.com/canonical/maas-ui/blob/server-side-search-and-filter/src/app/store/machine/reducers.test.ts#L347

You'll probably have to set the request id by mocking nanoid:

https://github.com/canonical/maas-ui/blob/server-side-search-and-filter/src/app/kvm/components/KVMResourcesCard/KVMResourcesCard.test.tsx#L19

Reply:
I ended up removing the custom spy for for `nanoid` (`jest.spyOn(reduxToolkit, "nanoid").mockReturnValue("mocked-nanoid");`) and using `useState` for `requestId`.

Mocking the nanoid in `hooks.test.tsx` was altering the internal logic of the hook and hiding issues with the implementation - `useRef` doesn't trigger re-renders and the value for `loading` was never recomputed as it was reading from the ref value 
``` 
const loaded = useSelector((state: RootState) =>
  machineSelectors.detailsLoaded(state, requestId.current)
);
```